File: Data Structures & Algorithms/is-palindrome/submission-4.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int l = 0;
        int r = s.Count()-1;
        while(l<r){
            while(l<s.Count()-1 && !isCharacterOrDigit(s[l])) l++;
            while(r>-1 && !isCharacterOrDigit(s[r])) r--;
            if(isCharacterOrDigit(s[l]) && isCharacterOrDigit(s[r]) && char.ToLower(s[l])!=char.ToLower(s[r])){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }

    public bool isCharacterOrDigit(char a){
        if(((int)a>=65 && (int)a<=90)||
        ((int)a>=97 && (int)a<=122)||
        ((int)a>=48 && (int)a<=57)){
            return true;
        }
        return false;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-4.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<Map<Character, Integer>, List<Integer>> mp = new HashMap<>();
        for(String str:strs){
            Map<Character, Integer> count = new HashMap<>();
            for(char c:str.toCharArray()) count.put(c, count.getOrDefault(c, 0)+1);
            mp.computeIfAbsent(count, k -> new ArrayList()).add(str);
        }
        return new ArrayList(mp.values());
    }
}
